### PR TITLE
first cut on directions on running Spinnaker on gcp using docker compose

### PR DESCRIPTION
**Pull request by tomaslin**
instructions on how to get docker-compose to run on gcp @spinnaker/reviewers please take a look

### CODE REVIEW

**Comment by duftler** [comment]
I'll review this unless someone else really, really wants to (been meaning to have a go at docker compose anyway).

**Comment by tomaslin** [comment]
that will be great. I did run into an issue where files copied over by gcloud were not visible to compose, so might end up changing the path for the config depending on which user the docker machine setup uses in gcp.

File: experimental/docker-compose/README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Spinnaker and Docker Compose 
+# Spinnaker on Docker Compose 
 
 This is an experimental integration of Spinnaker and Docker Compose using docker images for spinnaker published on the Docker Hub.
```
Comment (duftler) [inline experimental/docker-compose/README.md]: s/spinnaker/Spinnaker

File: experimental/docker-compose/README.md [hunk 2]
```diff
@@ -16,13 +16,15 @@ These instructions have been tested on MacOS 10.11.1 and Docker Toolbox 1.9.0b
...
-Make sure you're happy with the configuration of your spinnaker-local.yml file at ../../config. The instructions to change this are in the file default-spinnaker-local.yml
+Install the [Docker ToolBox](https://www.docker.com/docker-toolbox)
 
-Make sure you have followed the setup needed at the Cloud level from the directions at the root of this project's README file. 
+Navigate to the config directory at the root of this project ( two directories up from the experimental folder ). Copy the `default-spinnaker-local.yml` file into a file called `spinnaker-local.yml`. Edit spinnaker-local.yml to fit your configuration.
 
-Install the [Docker ToolBox](https://www.docker.com/docker-toolbox)
+If you're using aws, create a directory called aws under the config directory. Copy over your credentials from the .aws directory into this directory. If you don't have a .aws directory, follow the instructions here to create those credentails. 
+
+If you're using Google Compute Platform, create a directory called gcp under the config directory. Copy your json credentails into this directory as gcp.json
```
Comment (duftler) [inline experimental/docker-compose/README.md]: s/credentails/credentials

File: experimental/docker-compose/README.md [hunk 3]
```diff
@@ -16,13 +16,15 @@ These instructions have been tested on MacOS 10.11.1 and Docker Toolbox 1.9.0b
...
 ## Requirements
 
-Make sure you're happy with the configuration of your spinnaker-local.yml file at ../../config. The instructions to change this are in the file default-spinnaker-local.yml
+Install the [Docker ToolBox](https://www.docker.com/docker-toolbox)
 
-Make sure you have followed the setup needed at the Cloud level from the directions at the root of this project's README file. 
+Navigate to the config directory at the root of this project ( two directories up from the experimental folder ). Copy the `default-spinnaker-local.yml` file into a file called `spinnaker-local.yml`. Edit spinnaker-local.yml to fit your configuration.
 
-Install the [Docker ToolBox](https://www.docker.com/docker-toolbox)
+If you're using aws, create a directory called aws under the config directory. Copy over your credentials from the .aws directory into this directory. If you don't have a .aws directory, follow the instructions here to create those credentails.
```
Comment (duftler) [inline experimental/docker-compose/README.md]: s/credentails/credentials

File: experimental/docker-compose/README.md [hunk 4]
```diff
@@ -16,13 +16,15 @@ These instructions have been tested on MacOS 10.11.1 and Docker Toolbox 1.9.0b
 
 ## Requirements
 
-Make sure you're happy with the configuration of your spinnaker-local.yml file at ../../config. The instructions to change this are in the file default-spinnaker-local.yml
+Install the [Docker ToolBox](https://www.docker.com/docker-toolbox)
 
-Make sure you have followed the setup needed at the Cloud level from the directions at the root of this project's README file. 
+Navigate to the config directory at the root of this project ( two directories up from the experimental folder ). Copy the `default-spinnaker-local.yml` file into a file called `spinnaker-local.yml`. Edit spinnaker-local.yml to fit your configuration.
```
Comment (duftler) [inline experimental/docker-compose/README.md]: s/two directories up/one directory up

File: experimental/docker-compose/README.md [hunk 5]
```diff
@@ -16,13 +16,15 @@ These instructions have been tested on MacOS 10.11.1 and Docker Toolbox 1.9.0b
...
-Make sure you're happy with the configuration of your spinnaker-local.yml file at ../../config. The instructions to change this are in the file default-spinnaker-local.yml
+Install the [Docker ToolBox](https://www.docker.com/docker-toolbox)
 
-Make sure you have followed the setup needed at the Cloud level from the directions at the root of this project's README file. 
+Navigate to the config directory at the root of this project ( one directory up from the experimental folder ). Copy the `default-spinnaker-local.yml` file into a file called `spinnaker-local.yml`. Edit spinnaker-local.yml to fit your configuration.
 
-Install the [Docker ToolBox](https://www.docker.com/docker-toolbox)
+If you're using aws, create a directory called aws under the config directory. Copy over your credentials from the .aws directory into this directory. If you don't have a .aws directory, follow the instructions here to create those credentials. 
+
+If you're using Google Compute Platform, create a directory called gcp under the config directory. Copy your json credentials into this directory as gcp.json
```
Comment (duftler) [inline experimental/docker-compose/README.md]: The `gcp` and `aws` directories either need to be siblings to the `config` directory or the `volumes` need to be adjusted:
```
  volumes:
    - "../../config:/opt/spinnaker/config"
    - "../../aws:/root/.aws"
    - "../../gcp:/root/.gcp"
```
Creating them under the `config` directory won't work exactly as-is.

Comment (duftler) [inline experimental/docker-compose/README.md]: Also, make sure to set `jsonPath` to `/root/.gcp/gcp.json`.

File: experimental/docker-compose/README.md [hunk 6]
```diff
@@ -39,6 +41,76 @@ Under Kitematic, click on the application 'deck' -> settings -> ports. Click on
...
+
+One of the trickier bits of setting up spinnaker is to get the cloud providers configured correctly. It's recommended that you test out your configuration first on your local machine and then push to the cloud once you're happy with the results. 
+
+If you don't already have a Google Compute Platform account, you can create one [here](https://cloud.google.com/compute/). 
+
+Create a project that will hold your Spinnaker install and keep track of the project id ( which might be different than the project name ).
+
+## Set up Docker Machine Environment for GCP
+
+If you haven't already, obtain and set your CCP credentials following [these instructions](https://developers.google.com/identity/protocols/application-default-credentials#howtheywork).
```
Comment (duftler) [inline experimental/docker-compose/README.md]: s/CCP/GCP

File: experimental/docker-compose/README.md (hunk 5 above)
Comment (duftler) [inline experimental/docker-compose/README.md]: The instructions still say to create `aws` and `gcp` dirs under `config` instead of next to it.

**Pull request merged**